Fix install/kubernetes update-versions make target

joestringer opened the pull request:
Rework the `make -C install//kubernetes update-versions` make target to properly inject image tags into all the new Helm locations introduced by #13259 .

Tested by manually changing the `VERSION` file to test out `1.9-dev`, `1.8.5` and `1.9.90` and eyeballing the YAML changes.

Review commit-by-commit.

joestringer committed:
install: Fix version update scripting

Fix the Makefile to properly inject the appropriate image tags into the
various Cilium values fields during the release bump process.

Remove the redundant comments explaining repository, tag and pullPolicy
since they would interfere with the sed expression here that matches the
line after the repository line to update the tag.

Signed-off-by: Joe Stringer <joe@cilium.io>

joestringer committed:
helm: Bump dependency versions

Signed-off-by: Joe Stringer <joe@cilium.io>

joestringer commented on install/kubernetes/Makefile:
```diff
@@ -39,16 +44,29 @@ update-versions:
+		done;											\
+		sed -i 's/'$(CILIUM_PULLPOLICY_REGEX)'/\1 '$$pull_policy'/' $$chart;			\
+	done
+	$(QUIET)yq w -i $(CILIUM_VALUES) etcd.image.tag $(MANAGED_ETCD_VERSION)
```
This introduces `yq` as a dependency, maybe should consider adding something earlier in the target to check for its presence..? Wasn't sure the canonical way to add a CLI tool check.

pchaigno replied:
I couldn't find a canonical way or even another example in our code base, but maybe the following would do the trick:
```makefile
update-versions:
    @if ! yq --version &> /dev/null ; then echo "Please install yq"; exit 1; fi
```

aanm replied:
I can't find the package for my distribution that has `yq`, can we do this with a docker image? https://github.com/mikefarah/yq#run-with-docker

kaworu replied:
A clean way to achieve that in make is to define a `.PHONY` target that check whether `jq` is installed and have it as a dependency, .e.g https://github.com/cilium/cilium/blob/v1.8/api/v1/Makefile#L51-L53.

joestringer replied:
Cool, I'll keep that tucked behind my ear for next time it comes up.

pchaigno commented on install/kubernetes/Makefile:
```diff
@@ -39,16 +44,29 @@ update-versions:
+	done
+	$(QUIET)yq w -i $(CILIUM_VALUES) etcd.image.tag $(MANAGED_ETCD_VERSION)
+	$(QUIET)yq w -i $(CILIUM_VALUES) hubble.ui.proxy.image.tag $(HUBBLE_PROXY_VERSION)
+	$(QUIET)yq w -i $(CILIUM_VALUES) nodeinit.image.tag $(NODEINIT_VERSION)
```
My `yq` doesn't seem to know these `w` commands... I installed it using `pip3`.
```
$ yq --version
yq 2.11.1
```

kaworu commented on install/kubernetes/cilium/values.yaml:
```diff
@@ -42,141 +40,121 @@ serviceAccounts:
+#   mountPath: /srv/txt_collector
+#   hostPath: /var/lib/agent
+#   readOnly: true
+#   mountPropagation: HostToContainer
```
Theses comments diff are annoying as my understanding is that the commented options should be "under" `extraHostPathMounts` (i.e. not at the same level), which is "obvious" before this patch but not after.

joestringer replied:
Thanks for the feedback, I didn't put much thought into this since the yq version I have automatically makes this kind of change. But it sounds like I should put some more effort into finding out if there's a less intrusive implementation that tolerates whitespace in a better way.

joestringer committed:
install: Add workaround for helm chart versions

This overrides the version in the tree so that when you run `make -C
install/kubernetes`, it doesn't break the YAML generation for the
values.yaml and quick-install.yaml with an upcoming patch.

When we branch for v1.9 (circa v1.9.0-rc2), we can drop this.

Signed-off-by: Joe Stringer <joe@cilium.io>

joestringer committed:
install: Make shell invocation more obvious

This particular statement was being used to iterate over a single item
in a variable which is a particular filepath, but really what is needed
is a shell invocation to run multiple commands with shell syntax. Modify
it to make this more clear.

Signed-off-by: Joe Stringer <joe@cilium.io>

fristonio reviewed (approved):
LGTM
